### DISCUSSION TITLE
Lesson 12: Having problems with the OpenZeppelin-contracts Package

### DISCUSSION THREAD

**Discussion by mrthedude** [Cyfrin/foundry-full-course-cu#835] [Q&A]
I am using the Hardhat Nomic Foundation Solidity extension (same as Patrick's) but after I import the OpenZeppelin contracts package, I get an error on VS Code saying that it can't ```find local hardhat module``` and to check if I am missing any dependencies. I looked through the extension page and didn't really find any mention of dependencies I need to have to avoid an error like this.

I don't have any other compiler extensions and I uninstalled then reinstalled the Hardhat one but no luck resolving the issue. I'm attaching pics of the error and of the extensions I do have for added context.

Thankyou in advance for taking the time to help!
![Screenshot 2023-09-09 171804](https://github.com/Cyfrin/foundry-full-course-f23/assets/95384358/e9b133c0-f958-4268-8c02-6dc128e478d6)

![Screenshot 2023-09-09 172220](https://github.com/Cyfrin/foundry-full-course-f23/assets/95384358/f2ced1ff-09ce-4f1f-9c13-8b4ce64ac3e6)

**Comment by Niferu** [comment]
Hello @mrthedude 

I'm receiving this error too but it doesnt effect anything while working on ubuntu. Please make sure you have below in your `foundry.toml`.

```toml
remappings = ["@openzeppelin/contracts=lib/openzeppelin-contracts/contracts"]
```

**Reply by mrthedude** [reply]
Yep! I made sure that my remapping was correct in my ```foundry.toml``` too, sorry I should have mentioned that 😅 . I haven't really noticed any effects either, the ```ERC20Mock``` contract that Patrick uses in the course has since been updated in the OpenZeppelin contracts package but no issues running the package. It's weird though that the warning says to check for extension dependencies

**Comment by theirrationalone** [comment]
Please refer to [this](https://github.com/Cyfrin/foundry-full-course-f23/discussions/804#discussioncomment-6929243)

**Reply by mrthedude** [reply]
No change unfortunately. First I tried to replace my ```foundry.toml``` dependency with the text you provided, but the error persisted. Then I tried added a new ```txt``` file with your provided text and left my ```foundry.toml``` unchanged, but this didn't work either . As you said, there isn't any noticeable functionality problems but the ```dependencies error``` is still there.

Here's a copy of my ```foundry.toml``` before any changes:
```
[profile.default]
src = "src"
out = "out"
libs = ["lib"]

remappings = [
    "@chainlink/contracts=lib/chainlink-brownie-contacts/contracts/",
    "@openzeppelin/contracts=lib/openzeppelin-contracts/contracts",
]
````
Here's a picture of me replacing my original ```foundry.toml``` with your provided text (NOTE: the error was still here, I just didn't capture the info box in the picture)
![Screenshot 2023-09-11 101123](https://github.com/Cyfrin/foundry-full-course-f23/assets/95384358/5ed70675-9cbb-4f48-b0ac-311197926599)

Here's a picture of me keeping my original ```foundry.toml``` but having a ```txt``` file with your provided text 
![Screenshot 2023-09-11 101454](https://github.com/Cyfrin/foundry-full-course-f23/assets/95384358/1d17dae9-fb73-4ce2-9b51-8f67e5493b3b)

**Reply by Niferu** [reply]
@mrthedude 

Im keep getting this error on random timing and reloading window helps to solve it, but as I already mentioned it does not affect anything and you can simply close this popup window. This error is probably because we are using hardhat extension to work with foundry and we are not using hardhat tools that is why extension is yelling at us, but we can ignore it or do below:

Click `ctrl+shift+p`
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/110537615/9e1ec541-e267-46c4-80e6-b48cc97b33df)

**Reply by mrthedude** [reply]
Make sense, thankyou for taking the time to help!
